docs/user/azure/install_upi.md

```diff
@@ -0,0 +1,559 @@
+# Install: User Provided Infrastructure (UPI)
+
+The steps for performing a user-provided infrastructure install are outlined here. Several
+[Azure Resource Manager][azuretemplates] templates are provided to assist in
+completing these steps or to help model your own. You are also free to create
+the required resources through other methods; the templates are just an
+example.
+
+## Prerequisites
+
+* all prerequisites from [README](README.md)
+* the following binaries installed and in $PATH:
+  * [openshift-install][openshiftinstall]
+  * [az (Azure CLI)][azurecli] installed and aunthenticated
+  * python3
+  * [jq][jqjson]
+  * [yq][yqyaml]
+* python dotmap library: install it with `pip install dotmap`
+
+## Create an install config
+
+Create an install configuration as for [the usual approach](install.md#create-configuration):
+
+```console
+$ openshift-install create install-config
+? SSH Public Key /home/user_id/.ssh/id_rsa.pub
+? Platform azure
+? azure subscription id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+? azure tenant id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+? azure service principal client id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+? azure service principal client secret xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+INFO Saving user credentials to "/home/user_id/.azure/osServicePrincipal.json"
+? Region centralus
+? Base Domain example.com
+? Cluster Name test
+? Pull Secret [? for help]
+```
+
+### Extract data from install config
+
+Some data from the install configuration file will be used on later steps. Export them as environment variables with:
+
+```sh
+export CLUSTER_NAME=`yq -r .metadata.name install-config.yaml`
+export AZURE_REGION=`yq -r .platform.azure.region install-config.yaml`
+export SSH_KEY=`yq -r .sshKey install-config.yaml | xargs`
+export BASE_DOMAIN=`yq -r .baseDomain install-config.yaml`
+export BASE_DOMAIN_RESOURCE_GROUP=`yq -r .platform.azure.baseDomainResourceGroupName install-config.yaml`
+export RESOURCE_GROUP=$CLUSTER_NAME
```

Comment:
maybe use the metadata.json to extract the infra ID and use that to create the resource group?

Reply:
the metadata.json is available as part of `create ignition-configs`

Reply:
we can skip this comment as https://github.com/openshift/installer/pull/2822/files#diff-13c013f3cbb8bfed4da84d204129e1feR99 updates the manifests and we can't use this information...